src/services/deployment.ts

```diff
@@ -130,32 +131,46 @@ export const getDeploymentAfter28Days = async (
 ): Promise<Deployment[]> => {
   const PAGE = 1;
   const PER_PAGE = 1;
-  const environments = await getProjectEnvironments(projectId, groupToken);
-  const getDeploymentsPromises = environments.reduce<Promise<{ data: Deployment[]; headers: Headers }>[]>(
-    (deploymentsPromises, currentEnvironment) => {
-      if (
-        currentEnvironment.tier === EnvironmentTier.PRODUCTION ||
-        (isSendStagingEventsEnabled() && currentEnvironment.tier === EnvironmentTier.STAGING)
-      ) {
-        deploymentsPromises.push(
-          getProjectRecentDeployments(PAGE, PER_PAGE, {
-            groupToken,
-            projectId,
-            environmentName: currentEnvironment.name,
-            dateAfter,
-            dateBefore,
-          }),
-        );
-      }
-
-      return deploymentsPromises;
-    },
-    [],
-  );
+  try {
+    const environments = await getProjectEnvironments(projectId, groupToken);
+    const getDeploymentsPromises = environments.reduce<Promise<{ data: Deployment[]; headers: Headers }>[]>(
+      (deploymentsPromises, currentEnvironment) => {
+        if (
+          currentEnvironment.tier === EnvironmentTier.PRODUCTION ||
+          (isSendStagingEventsEnabled() && currentEnvironment.tier === EnvironmentTier.STAGING)
+        ) {
+          deploymentsPromises.push(
+            getProjectRecentDeployments(PAGE, PER_PAGE, {
+              groupToken,
+              projectId,
+              environmentName: currentEnvironment.name,
+              dateAfter,
+              dateBefore,
+            }),
+          );
+        }
+
+        return deploymentsPromises;
+      },
+      [],
+    );
 
-  const promisesResponse = await Promise.all(getDeploymentsPromises);
```

Comment:
A single error here would have bubbled up. It looks like the new code is silently catching errors and returning an empty array. Do we want to change that behavior?

Reply:
Yeah, maybe better throw error in `catch` block, or remove `try catch`, because `getDeploymentAfter28Days` is used in `hasDeploymentAfter28Days`, and this method also is called in `Promise.allSettled` that is in `try catch` block. But the better option it's remove `try catch` block to stay the same logic